Web: Use Intl.ListFormat rather than .join(", ") #26936

CIC4DA opened the pull request:
This PR fixes a web issue by replacing the inconsistent .join(", ") with Intl.ListFormat for accurate and localized list formatting, using a common utility function.

fixes: Use Intl.ListFormat rather than .join(", ") #26936

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

khanumar03 commented:
@CIC4DA  I'm having  problem with merging while  pull request  here is the ss please  help me figure out how  can I  fix this
![Screenshot 2023-12-20 182721](https://github.com/zulip/zulip/assets/125285100/d73c72bc-52c3-4f5f-8427-03231c4cca43)

andersk commented:
- Some of these call sites absolutely must not be internationalized, such as when joining CSS selectors.

- `style: "narrow"` is not appropriate for every use. We need to check this on a case-by-case basis, but I think we will usually want the default `style: "long"`. The `type` may be important in some cases as well.

- Safari doesn’t support `Intl.ListFormat` on macOS 10.15 and earlier, so we need a fallback or polyfill.

CIC4DA commented:
Got it, thanks for the heads up! I'll start working on addressing these concerns.

Could you please specify the files or components where we need to be mindful of internationalization,
According to me we could use internationalization with  ```{style: long, type: conjuctions}``` in all other files I have changed except in reactions and realm_playground.

If you could update me about this, I will make the required changes in PR with fallback for macOS.

CIC4DA commented:
@andersk, Could you please review it again.

I have resolved the following issues:

- [ ]  Add a fallback system for MacOs.
- [ ]  Files which were not supposed to be internationalized are reverted back to using .join(", ").
- [ ]  Introduced ```{ style: "long", long: "conjunction" }``` could be changed according to use.
- [ ]  Refactor tests to test with the new changes.

CIC4DA commented:
@andersk could you please review my PR and let me know if there are any tweaks needed,
Thanks.

CIC4DA commented:
Since this PR contained a lot of commits, I have resolved that and made a new PR #28453, for this issue.
